src/alarm_trap_sender.cpp

```diff
@@ -143,3 +166,112 @@ void AlarmTrapSender::send_trap(const AlarmTableDef& alarm_table_def)
+  memset(&var_alarm_description, 0x00, sizeof(var_alarm_description));
+  memset(&var_alarm_details, 0x00, sizeof(var_alarm_details));
+  memset(&var_alarm_cause, 0x00, sizeof(var_alarm_cause));
+  memset(&var_alarm_effect, 0x00, sizeof(var_alarm_effect));
+  memset(&var_alarm_action, 0x00, sizeof(var_alarm_action));
+
+  var_trap.next_variable = &var_MIB_version;
+  var_MIB_version.next_variable = &var_alarm_name;
+  var_alarm_name.next_variable = &var_alarm_oid;
+  var_alarm_oid.next_variable = &var_resource_id;
+  var_resource_id.next_variable = &var_alarm_severity;
+  var_alarm_severity.next_variable = &var_alarm_description;
+  var_alarm_description.next_variable = &var_alarm_details;
+  var_alarm_details.next_variable = &var_alarm_cause;
+  var_alarm_cause.next_variable = &var_alarm_effect;
+  var_alarm_effect.next_variable = &var_alarm_action;
+  var_alarm_action.next_variable = NULL;
+
+  snmp_set_var_objid(&var_trap, snmp_trap_oid, OID_LENGTH(snmp_trap_oid));
+  snmp_set_var_objid(&var_MIB_version, MIB_version_oid, OID_LENGTH(MIB_version_oid));
+  snmp_set_var_objid(&var_alarm_name, alarm_name_oid, OID_LENGTH(alarm_name_oid));
+  snmp_set_var_objid(&var_alarm_oid, alarm_oid_oid, OID_LENGTH(alarm_oid_oid));
+  snmp_set_var_objid(&var_resource_id, resource_id_oid, OID_LENGTH(resource_id_oid));
+  snmp_set_var_objid(&var_alarm_severity, alarm_severity_oid, OID_LENGTH(alarm_severity_oid));
+  snmp_set_var_objid(&var_alarm_description, alarm_description_oid, OID_LENGTH(alarm_description_oid));
+  snmp_set_var_objid(&var_alarm_details, alarm_details_oid, OID_LENGTH(alarm_details_oid));
+  snmp_set_var_objid(&var_alarm_cause, alarm_cause_oid, OID_LENGTH(alarm_cause_oid));
+  snmp_set_var_objid(&var_alarm_effect, alarm_effect_oid, OID_LENGTH(alarm_effect_oid));
+  snmp_set_var_objid(&var_alarm_action, alarm_action_oid, OID_LENGTH(alarm_action_oid));
+
+  snmp_set_var_typed_value(&var_trap, ASN_OBJECT_ID, 
+                           (u_char*) trap_type_oid, 
+                           sizeof(trap_type_oid));
+  snmp_set_var_typed_value(&var_MIB_version, ASN_OCTET_STR,
+                           MIB_version.c_str(),
+                           MIB_version.length());
+  snmp_set_var_typed_value(&var_alarm_name, ASN_OCTET_STR,
+                           alarm_table_def.name().c_str(),
+                           alarm_table_def.name().length());
+  snmp_set_var_typed_value(&var_alarm_oid, ASN_OBJECT_ID,
+                           (u_char*) model_row_oid,
+                           sizeof(model_row_oid));
+
+  var_alarm_oid.val.objid[ALARMMODELTABLEROW_INDEX] = alarm_table_def.alarm_index();
+  
+  snmp_set_var_typed_value(&var_resource_id, ASN_OBJECT_ID, 
+                           (u_char*) zero_dot_zero, 
+                           sizeof(zero_dot_zero));
+  snmp_set_var_typed_value(&var_alarm_severity, ASN_OCTET_STR,
+                           std::to_string(severity_to_state[alarm_table_def.severity()]).c_str(),
```

Comment:
At the moment this sets the severity to the string '1' (or the string form of some other number). That's not quite what I'd have expected - if it's a string I'd expect it to be something like "CRITICAL", if it's a number I'd expect it to be ASN_UNSIGNED and have some kind of enumeration defined in the MIB file.

@eleanor-merry - I think this is a spec question for you?